source/common/quic/envoy_quic_client_session.cc

```diff
@@ -25,10 +30,18 @@ class EnvoyQuicProofVerifyContextImpl : public EnvoyQuicProofVerifyContext {
     return transport_socket_options_;
   }
 
+  absl::string_view getEchNameOverrride() const override {
+    const char* name = nullptr;
+    size_t name_len = 0;
+    SSL_get0_ech_name_override(ssl_info_.ssl(), &name, &name_len);
```

Comment:
Wait, are you all adding support for ECH already? This is only relevant if you're supporting the experimental ECH extension.
https://datatracker.ietf.org/doc/html/draft-ietf-tls-esni-14

I do intend to get ECH into QUICHE at some point, but I wasn't aware that Envoy was part of the experiment plan. Are you all calling other ECH APIs? Are you all following the standards process? Note that both the protocol and the API are, of course, experimental as they may evolve over time.

Reply:
Envoy isn't intentionally experimenting it. But this PR tries to make DefaultCertValidator to do whatever ssl_crypto_x509_session_verify_cert_chain() https://github.com/google/boringssl/blob/8ba90d18175ed1b591c16ddedc95f5c51da56d22/ssl/ssl_x509.cc#L380] is doing, even though some of the boring SSL feature is not enabled. ssl_crypto_x509_session_verify_cert_chain() is called implicitly in boring SSL with the old SSL_CTX_set_cert_verify_callback() interface, but not longer called after switching to SSL_CTX_set_custom_verify().

Reply:
If you're not experimenting with it, you shouldn't be calling this yet. Things may change and if ECH isn't configured, it will never return anything. Indeed if you're copying BoringSSL's code, that code may change over time anyway, so that's not really a sound strategy. Envoy is ultimately responsible for its implementation of that callback.

Reply:
We can make this return empty string for now. But once you get ECH into QUICHE, we will need to come back to implement this interface. Does this sounds right?